x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/UnaryScalarFunction.java

```diff
@@ -19,6 +19,10 @@
 
 public abstract class UnaryScalarFunction extends ScalarFunction {
 
+    public interface UnaryNegateable {
```

Comment:
Regardless of the number of children, there should be a `negatable` (I believe that's the correct syntax) interface common to all. In other words, Negatable should be moved up from `BinaryOperator` to potentially a standalone interface and likely have its generic signature extended to bound its return type (`Negatable<T extends ScalarFunction>`).